src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6377,7 +6377,9 @@ public override IOperation VisitIncrementOrDecrement(IIncrementOrDecrementOperat
 
         public override IOperation VisitDiscardOperation(IDiscardOperation operation, int? captureIdForResult)
         {
-            return new DiscardOperation(operation.DiscardSymbol, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
+            return new DiscardOperation(
+                operation is IPatternOperation pat ? pat.InputType : operation.DiscardSymbol?.Type,
```

Comment:
>operation is IPatternOperation pat ? pat.InputType : operation.DiscardSymbol?.Type, [](start = 16, length = 83)

In the `Type _` pattern, shouldn't the discard already get the proper type (ie. `Type`)? I feel we shouldn't have to look at the pattern.
I think this connects to Fred's question: why have a node that is both an IDiscardOperation and an IPatternOperation?

Reply:
What was the resolution for this?
If you want to postpone resolution to the next PR, would it make sense to add a PROTOTYPE comment?

---
In reply to: [244028898](https://github.com/dotnet/roslyn/pull/31967#discussion_r244028898) [](ancestors = 244028898)